docs/batch-local-execution-mode.md

```diff
@@ -0,0 +1,90 @@
+# Design: Local execution mode for online serving queries
+
+## Background
+
+The query engine of risingwave is expected to support two kinds of query: highly concurrent point query and adhoc query.
+The characteristics of these two different kinds of queries are summarized as followings:
+
+|	|Point Query	|Adhoc query	|
+|---	|---	|---	|
+|Latency	|several ms	|ms to minutes	|
+|QPS	|1000 ~ 100000	|100 ~ 1000	|
+|SQL	|Simple	|Arbitrary complex	|
+|Result Set	|Small	|Small, Medium, Large	|
+|Use Case	|Dashboard	|Engineers, Data Scientist	|
+
+Our distributed query processing engine is designed for complex adhoc query, and it can’t meet the latency/qps 
+requirement of point query, and in this article we introduce local execution mode for point query.
+
+## Design
+
+![Frontend Flow](./images/batch-local-execution-mode/frontend-flow.svg)
+
+## Example 1: select a from t where b in (1, 2, 3, 4)
+
+Let’s use above sql as example:
+
+
+![Example 1](./images/batch-local-execution-mode/example1.svg)
+
+The key changes from distributed mode:
+
+1. The exchange executor will be executed directly by local query execution, not by distributed scheduler. This means that we no longer have async execution/monitoring, etc.
+2. The rpc is issued by exchange executor directly, not by scheduler.
+
+## Example2: SELECT pk, t1.a, t1.fk, t2.b FROM t1, t2 WHERE t1.fk = t2.pk AND t1.pk = 114514
+
+Following is the plan and execution of above sql in local mode:
+
+
+![Example 2](./images/batch-local-execution-mode/example2.svg)
+
+As explained above, the lookup join/exchange phase will be executed directly on frontend. The pushdown(filter/table, both the build and probe side) will be issued by executors rather than scheduler.
+
+### Optimization/Scheduling
+
+The overall process will be quite similar to distributed processing, but with a little difference:
+
+1. We only use heuristic optimizater for it, and only a limited set of rules will be applied.
+2. No scheduler will involve, and the physical plan is executed in current thread(coroutine) immeidately.
+
+### Monitoring/Management
+
+Local execution mode will not go through query management mentioned in  batch query manager to reduce latency as 
+much as possible.
+
+### How to switch between local/distributed execution mode
```

Comment:
-> ... local and distributed execution modes?
Please add a question mark at the end.